chain/network/src/store/schema/mod.rs

```diff
@@ -101,6 +101,36 @@ impl BorshRepr for KnownPeerStateRepr {
     }
 }
 
+/// A Borsh representation of the primitives::ConnectionInfo.
+#[derive(BorshSerialize, BorshDeserialize)]
+pub struct ConnectionInfoRepr {
+    peer_info: primitives::PeerInfo,
+    /// UNIX timestamps in nanos.
+    first_connected: u64,
+    last_connected: u64,
+}
```

Comment:
I am not completely satisfied with `first_connected` either, but we may not have been continuously connected to this peer which `connected_since` and `connected_until` seem to imply.

Reply:
ow, my intention was to actually store exactly the intervals of continuous connection. So in case a node gets connected/disconnected a few times, then each interval of being connected counts separately.

Reply:
> Should the interval be broken when such an event occurs?
Yes, in a sense that if you have long connection interval [b0,e0] and then short one [b1,e1]  (with a given peer) such that e0 < b1, then
1. first you store a reliable connection interval [b0,e0]
2. upon disconnection you try to reconnect
3. since the new connection survived only [b1,e1] it is not stored, but [b0,e0] is not removed
4. you still try to reconnect, until [b0,e0] is not removed from RC by some newer reliable connection

The intended semantics here is that a reliable connections may become unreliable in a sense, that it will break frequently. Then we should NO LONGER consider it reliable, even though we are "often" connected to (because we reconnected), just not continuously.